sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java

```diff
@@ -346,8 +355,14 @@ private void processEvents(PartitionContext partitionContext, PartitionProcessor
                     ? partitionEventBatch.get(0).getData() : null);
                 LastEnqueuedEventProperties lastEnqueuedEventProperties = (partitionEventBatch.size() == 1
```

Comment:
Ah yeah, the reason we have `lastEnqueuedEventProperties[0]` is that in a lambda, the reference has to be effectively final.

Reply:
Got it.  Or we can put updateOrGetLastEnqueuedEventProperties() inside the lambda, it is readable but have extra cost of function calls.

Reply:
> we can put updateOrGetLastEnqueuedEventProperties() inside the lambda

After thinking more, this change has different behavior. It will update last event properties base on each partition event. Currently implementation only check the last `partitionEvent`'s properties inside `partitionEventBatch` and update if it is not null. Please verify which one is correct.